File: fly/src/main/java/com/xia/fly/base/delegate/FragmentDelegate.java
package com.xia.fly.base.delegate;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.View;

/**
 * {@link Fragment} 代理类,用于框架内部在每个 {@link Fragment} 的对应生命周期中插入需要的逻辑
 *
 * @author xia
 * @date 2018/9/20.
 * @see FragmentDelegateImpl
 * @see <a href="https://github.com/JessYanCoding/MVPArms/wiki#3.13">FragmentDelegate wiki 官方文档</a>
 */
public interface FragmentDelegate {

    String FRAGMENT_DELEGATE = "FRAGMENT_DELEGATE";

    void onAttach(@NonNull Context context);

    void onCreate(@Nullable Bundle savedInstanceState);

    void onCreateView(@Nullable View view, @Nullable Bundle savedInstanceState);

    void onActivityCreate(@Nullable Bundle savedInstanceState);

    void onStart();

    void onResume();

    void onPause();

    void onStop();

    void onSaveInstanceState(@NonNull Bundle outState);

    void onDestroyView();

    void onDestroy();

    void onDetach();

    /**
     * Return true if the fragment is currently added to its activity.
     */
    boolean isAdded();
}

File: fly/src/main/java/com/xia/fly/base/delegate/FragmentDelegate.kt
package com.xia.fly.base.delegate

import android.content.Context
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment

/**
 * [Fragment] 代理类,用于框架内部在每个 [Fragment] 的对应生命周期中插入需要的逻辑
 * @see [FragmentDelegateImpl]
 * @see [FragmentDelegate wiki 官方文档](https://github.com/JessYanCoding/MVPArms/wiki#3.13)
 *
 * @author xia
 * @date 2018/9/20.
 */
interface FragmentDelegate {

    /**
     * Return true if the fragment is currently added to its activity.
     */
    fun isAdded(): Boolean

    fun onAttach(context: Context)

    fun onCreate(savedInstanceState: Bundle?)

    fun onCreateView(view: View?, savedInstanceState: Bundle?)

    fun onActivityCreate(savedInstanceState: Bundle?)

    fun onStart()

    fun onResume()

    fun onPause()

    fun onStop()

    fun onSaveInstanceState(outState: Bundle)

    fun onDestroyView()

    fun onDestroy()

    fun onDetach()

    companion object {

        const val FRAGMENT_DELEGATE = "FRAGMENT_DELEGATE"
    }
}
